tests/test_device_properties.py

```diff
@@ -0,0 +1,45 @@
+from nidaqmx.constants import BusType, TriggerUsage
+
+
+class TestDevicePropertyDataTypes(object):
+    """Contains a collection of pytest tests.
+
+    This validates the property getter,setter and deleter methods for different
+    data types of device properties.
+    """
+
+    def test_boolean_property(self, bridge_device):
+        """Test for validating the device property of boolean type."""
+        # The PXIe-4331 supports analog triggering
+        assert bridge_device.anlg_trig_supported
+
+    def test_float_property(self, bridge_device):
+        """Test for validating the device property of float type."""
+        # The ai max single channel rate of the NI PXIe-4331 device is 102400.0
+        assert bridge_device.ai_max_single_chan_rate == 102400.0
+
+    def test_uint_property(self, bridge_device):
+        """Test for validating the device property of uint type."""
+        assert bridge_device.product_num == 1957283012
+
+    def test_string_property(self, bridge_device):
+        """Test for validating a device property of string type."""
+        assert bridge_device.product_type == "PXIe-4331"
+
+    def test_enum_property(self, bridge_device):
+        """Test for validating a device property of enum type."""
+        assert bridge_device.bus_type == BusType.PXIE
+
+    def test_list_of_float_property(self, bridge_device):
```

Comment:
The properties which returned a list of float values where mostly properties which returned a range like current/voltage ranges and when attempting to read such properties the value returned was always an empty array. Hence the test also checks the same. 
Please let me know if there is a way to actual values with a simulated device.